sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -33,6 +36,7 @@
   }
 
 const char SupportedVersion[] = _PI_H_VERSION_STRING;
+std::map<pi_context, std::set<std::string>> SupportedExtensions;
```

Comment:
@MrSidims , we need to avoid any global state, and here it is trivial (and consistent) to keep this cache in the PI context object.

Reply:
Furthermore access to this var should be synchronized.

Reply:
If I understand correctly, you suggestion is to create a wrapper around OpenCL context and this set of extensions, right? In this case it would require significant rewriting of pi_opencl.cpp since there are direct mapping between PI API and OCL API (for example `_PI_CL(piContextRelease, clReleaseContext)`). I removed checking for supported extensions, leaving it to low-level runtime to decide whether a property shall be supported or not.

Reply:
> to create a wrapper around OpenCL context and this set of extensions, right?

I thought that wrapper existed already as I mistakenly assumed this was for Level-Zero plugin.
For OpenCL adding wrapper handlers for this is certainly an overkill.
So a global cache (map) would work, if you decided to add that filtering later.